numpy/_core/tests/test_mem_overlap.py

```diff
@@ -402,7 +402,7 @@ def check(A, U, exists=None):
             exists = (X is not None)
 
         if X is not None:
-            assert_(sum(a * x for a, x in zip(A, X)) == sum(a * u // 2 for a, u in zip(A, U)))
+            assert_(sum(a * x for a, x in zip(A, X)) == sum(a * u // 2 for a, u in zip(A, U)))  # noqa: E501
```

Comment:
I'd break this into two computations before the `assert_`. Now that we are using pytest, you can also use `assert` instead of the `assert_` function.